compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -453,8 +455,9 @@ pub fn maybe_create_entry_wrapper<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
                 .unwrap()
                 .unwrap(),
             );
-            let start_ty = cx.type_func(&[cx.val_ty(rust_main), isize_ty, i8pp_ty], isize_ty);
-            (start_fn, start_ty, vec![rust_main, arg_argc, arg_argv])
+            let start_ty =
+                cx.type_func(&[cx.val_ty(rust_main), isize_ty, i8pp_ty, bool_ty], isize_ty);
+            (start_fn, start_ty, vec![rust_main, arg_argc, arg_argv, arg_ignore_sigpipe])
```

Comment:
Please also update rustc_codegen_cranelift.

Reply:
I will do my best to update `rustc_codegen_cranelift`. Is there an easy way to run my `// run-pass` tests with cranelift to verify my implementation? You don't necessarily need to provide super detailed instructions, just some pointers would be greatly appreciated. Sorry for being a bit lazy and asking rather than trying to figure it out by myself.

Reply:
I'm afraid cranelift does not work on my system (Apple M1 Pro laptop). Trying to follow instructions at https://github.com/bjorn3/rustc_codegen_cranelift I get:

```
% ./y.rs build
[BUILD] y.rs
[BUILD] rustc_codegen_cranelift
    Finished release [optimized] target(s) in 0.10s
   Compiling core v0.0.0 (/Users/martin/src/rustc_codegen_cranelift/build_sysroot/sysroot_src/library/core)
   Compiling rustc-std-workspace-core v1.99.0 (/Users/martin/src/rustc_codegen_cranelift/build_sysroot/sysroot_src/library/rustc-std-workspace-core)
   Compiling compiler_builtins v0.1.74
[rustc_std_workspace_core      : codegen mono items] start
   Compiling libc v0.2.126
[rustc_std_workspace_core      : codegen mono items] end time: 1.092291ms
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: /Users/martin/src/rustc_codegen_cranelift/build_sysroot/target/aarch64-apple-darwin/release/deps/librustc_std_workspace_core-1b1fc330a6a81d7a.rlib(lib.rmeta) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: /Users/martin/src/rustc_codegen_cranelift/build_sysroot/target/aarch64-apple-darwin/release/deps/librustc_std_workspace_core-1b1fc330a6a81d7a.rlib(rustc_std_workspace_core-1b1fc330a6a81d7a.47gfowxiwf5bghsz.rcgu.o) has no symbols
warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: archive library: /Users/martin/src/rustc_codegen_cranelift/build_sysroot/target/aarch64-apple-darwin/release/deps/librustc_std_workspace_core-1b1fc330a6a81d7a.rlib the table of contents is empty (no object file members in the library define global symbols)
[core                          : codegen mono items] start
thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unimplemented relocation Relocation { offset: 32, size: 32, kind: Relative, encoding: Generic, symbol: SymbolId(43), addend: -4 }")', src/driver/aot.rs:58:38
```

and I don't think it is worthwhile for me to try to blindly fix cranelift. Is there any possibility you can help with this?

Btw, in

https://github.com/rust-lang/rust/blob/f19ccc2e8dab09e542d4c5a3ec14c7d5bce8d50e/compiler/rustc_codegen_cranelift/src/main_shim.rs#L145

I only see types, but no actual values that are passed. Are the values that are passed set up in some other place? Would you mind telling me where?

Reply:
Looks like AArch64 macOS support is broken. I don't test it on CI as GHA doesn't provide any runners for it. A best effort attempt at updating cg_clif for this PR is fine I guess. I can fix it up afterwards if you got it wrong.

> I only see types, but no actual values that are passed. Are the values that are passed set up in some other place? Would you mind telling me where?

`&[main_val, arg_argc, arg_argv]` is the argument value list.

Reply:
Sounds good, I'll do a best effort update then. Thank you.

> `&[main_val, arg_argc, arg_argv]` is the argument value list.

`arg_argc` and `arg_argv` is

https://github.com/rust-lang/rust/blob/f19ccc2e8dab09e542d4c5a3ec14c7d5bce8d50e/compiler/rustc_codegen_cranelift/src/main_shim.rs#L84-L85

So the same value is passed to both argc and argv? 🤔  And what value exactly? What value does `.pointer_type()` have?

Reply:
`append_block_param` is to add a function argument here. `.pointer_type()` is the type of a pointer, as the arguments are pointers.

Reply:
OK, so I think you only need to change one line that I could not figure out how to write. Namely this line:

```
// How? arg_ignore_sigpipe.value = _ignore_sigpipe;
```

Somehow `arg_ignore_sigpipe` needs to take the value of `_ignore_sigpipe` which is a regular `bool`.